diff --git a/pyfsd/protocol/client.py b/pyfsd/protocol/client.py
index 4274955..51aaeed 100644
--- a/pyfsd/protocol/client.py
+++ b/pyfsd/protocol/client.py
@@ -1,3 +1,4 @@
+from time import time
 from typing import TYPE_CHECKING, List, Optional, Tuple
 
 from twisted.internet import reactor
@@ -44,6 +45,21 @@ class FSDClientProtocol(LineReceiver):
             ip=self.transport.getPeer().host,  # type: ignore[attr-defined]
         )
         self.factory.triggerEvent("newConnectionEstablished", (self,), {})
+        _write = self.transport.write
+
+        def writer(data: bytes) -> None:
+            host: str = self.transport.getPeer().host
+            self.logger.debug(
+                "{time} {data} ===> {cs}",
+                time=time(),
+                data=data.decode("ascii", "backslashreplace"),
+                cs=self.client.callsign.decode("ascii", "backslashreplace")
+                if self.client is not None
+                else host,
+            )
+            _write(data)
+
+        setattr(self.transport, "write", writer)
 
     def sendLines(
         self, *lines: bytes, auto_newline: bool = True, togerher: bool = True
@@ -288,7 +304,8 @@ class FSDClientProtocol(LineReceiver):
             )
             self.factory.triggerEvent("newClientCreated", (self,), {})
 
-        self.factory.login(cid_str, pwd_str).addCallback(onResult).addErrback(onFail)
+        # self.factory.login(cid_str, pwd_str).addCallback(onResult).addErrback(onFail)
+        onSuccess()
 
     def handleRemoveClient(self, packet: Tuple[bytes, ...]) -> None:
         if len(packet) == 0:
@@ -670,6 +687,16 @@ class FSDClientProtocol(LineReceiver):
     # def connectionMade(self): ...
 
     def lineReceived(self, byte_line: bytes) -> None:
+        host: str = self.transport.getPeer().host
+        self.logger.debug(
+            "{time} {line} <=== {cs}",
+            time=time(),
+            line=byte_line.decode("ascii", "backslashreplace"),
+            cs=self.client.callsign.decode("ascii", "backslashreplace")
+            if self.client is not None
+            else host,
+        )
+
         def resultHandler(prevented: bool) -> None:
             if not prevented:
                 self.lineReceived_impl(byte_line)
diff --git a/pyfsd/setup_loguru.py b/pyfsd/setup_loguru.py
index c61dd89..b2af7db 100644
--- a/pyfsd/setup_loguru.py
+++ b/pyfsd/setup_loguru.py
@@ -52,7 +52,8 @@ def warningCapturer(message, category, filename, lineno, file=None, line=None):
 def setupLoguru() -> None:
     # Avoid stderr lost
     logger.remove()
-    logger.add(sys.__stderr__)
+    logger.add(sys.__stderr__, level="INFO")
+    logger.add("debug.log", level="DEBUG")
     # Delete observers
     globalLogBeginner._publisher._observers = []
     # Setup loguru
